code/game/mecha/equipment/weapons/melee.dm

```diff
@@ -8,64 +8,34 @@
 	var/dam_force = 20
 	required_type = /obj/mecha
 
-/obj/item/mecha_parts/mecha_equipment/melee_weapon/action(atom/target)
-	if(!action_checks(target)) return
-
-	if(isliving(target))
-		var/mob/living/M = target
-		if(M.stat > 1)
-			return
-		if(chassis.occupant.a_intent != I_HELP) // So that the help intent act as a kind of safety
-			M.take_overall_damage(dam_force)
-			M.updatehealth()
-			occupant_message(SPAN_WARNING("You attack [target] with [src.name]. Something cracks."))
-			chassis.visible_message(SPAN_WARNING("[chassis] attacked [target]."))
-			log_message("Attacked [target.name] with [name]")
-		else
-			step_away(M,chassis)
-			occupant_message("You push [target] out of the way.")
-			chassis.visible_message("[chassis] pushes [target] out of the way.")
-		set_ready_state(0)
-		chassis.use_power(energy_drain)
-		do_after_cooldown()
-		return 1
 
 /obj/item/mecha_parts/mecha_equipment/melee_weapon/sword
 	name = "mech sword"
-	desc = "A huge sword designed to be wielded by an exosuit. Somehow can't hit walls."
+	desc = "A huge sword designed to be wielded by an exosuit."
 	icon_state = "mech_sword"
+	sharp = TRUE
+	edge = TRUE
+	tool_qualities = list(QUALITY_CUTTING = 35, QUALITY_SAWING = 20) // It's a literal mech sized blade
```

Comment:
```suggestion
	tool_qualities = list(QUALITY_CUTTING = 60, QUALITY_SAWING = 45) // It's a literal mech sized blade
```
Armature cutters have 30 cutting & a basic metal saw has 35 sawing

Reply:
Not sure wether to accept it or not honestly, it is a giant sword BUT it's also not a precision tool, it's meant to kill and break stuff, not be better than a circular saw

Reply:
Fair enough

Reply:
I like wizards numbers for the time being.

Reply:
They're fine, yea. If we want to have real mech tools down the line with real tool qualities then that'd be doable

Reply:
If we go by the standardization of tool qualities:
15: makeshift tools, bare functional minimum (specially for crafting)
35: stock, basic everyday tools that are meant for this job in particular
+50: specialized tools that outperform normal tools by virtue of power/automatization/internal circuitry or hydraulics/etc

Then yes, I'd say the numbers are perfectly fair as they are right now, it's a mech sword, not a mech chainsaw designed specifically to cut down trees, and since it's huge, bulky and most probably sharp, it's justified to have higher cutting quality than a normal knife.